test/workbox-routing/node/Router.mjs

```diff
@@ -0,0 +1,302 @@
+import sinon from 'sinon';
+import {expect} from 'chai';
+
+import Route from '../../../packages/workbox-routing/lib/Route.mjs';
+import Router from '../../../packages/workbox-routing/lib/Router.mjs';
+import expectError from '../../../infra/utils/expectError';
+import {_private} from '../../../packages/workbox-core/index.mjs';
+
+describe(`[workbox-routing] Router`, function() {
+  const sandbox = sinon.sandbox.create();
+  const {logger} = _private;
+  const MATCH = () => {};
+  const HANDLER = () => {};
+  const METHOD = 'POST';
+  const EXPECTED_RESPONSE_BODY = 'test body';
+
+  beforeEach(async function() {
+    clearRequire.all();
+
+    // Run this in the `beforeEach` hook as well as the afterEach hook due to
+    // a mocha bug where `afterEach` hooks aren't run for skipped tests.
+    // https://github.com/mochajs/mocha/issues/2546
+    sandbox.restore();
+    // Prevent logs in the mocha output.
+    sandbox.stub(logger, 'warn');
+    sandbox.stub(logger, 'log');
+  });
+
+  afterEach(function() {
+    sandbox.restore();
+  });
+
+  describe(`constructor`, function() {
+    it(`should construct without any inputs`, function() {
+      expect(() => {
+        new Router();
+      }).to.not.throw();
+    });
+  });
+
+  describe(`calling addFetchListener()`, function() {
+    it(`should return true and add a fetch listener the first time addFetchListener() is called`, function() {
+      const spy = sandbox.spy(self, 'addEventListener');
+
+      const router = new Router();
+      const added = router.addFetchListener();
+
+      expect(added).to.be.true;
+      expect(spy.calledOnce).to.be.true;
+      expect(spy.firstCall.args[0]).to.eql('fetch');
+      expect(spy.firstCall.args[1]).to.be.a('function');
+    });
+
+    it(`should return false and not add a fetch listener the second time addFetchListener() is called`, function() {
+      const spy = sandbox.spy(self, 'addEventListener');
+
+      const router = new Router();
+      const addedFirst = router.addFetchListener();
+      const addedSecond = router.addFetchListener();
+
+      expect(addedFirst).to.be.true;
+      expect(addedSecond).to.be.false;
+      expect(spy.calledOnce).to.be.true;
+    });
+  });
+
+  describe(`registering and unregistering routes`, function() {
+    it(`should throw in dev when registerRoute is called with a parameter that doesn't match the expected interface`, async function() {
+      if (process.env.NODE_ENV === 'production') return this.skip();
+
+      const router = new Router();
+
+      await expectError(
+        () => router.registerRoute({}),
+        'missing-a-method',
+        (error) => {
+          expect(error.details).to.have.property('moduleName').that.eql('workbox-routing');
+          expect(error.details).to.have.property('className').that.eql('Router');
+          expect(error.details).to.have.property('funcName').that.eql('registerRoute');
+          expect(error.details).to.have.property('paramName').that.eql('route');
```

Comment:
Is there any way to expect a specific method name to be missing (Just thinking if I got this error and it didn't call out the method that was missing, it could be frustrating).

Reply:
Yeah, I'll switch.